src/main/java/be/seeseemelk/mockbukkit/inventory/ItemFactoryMock.java

```diff
@@ -94,13 +94,13 @@ public boolean isApplicable(ItemMeta meta, Material material)
 	@Override
 	public boolean equals(ItemMeta meta1, ItemMeta meta2)
 	{
-		if (meta1 != null && meta2 != null)
+		if (meta1 == null)
 		{
-			return meta1.equals(meta2);
+			return meta2 == null;
 		}
 		else
 		{
-			return false;
+			return meta1.equals(meta2);
 		}
```

Comment:
This can probably be simplified to
`return Objects.equals(meta1, meta2);`
which does null checks too and returns true if both are null.

Reply:
Also a comment explaining this behaviour would be nice :)